Some imported modules are missing type declarations

**Describe the bug**
```
> matrix-bifrost@0.1.8 build /media/d/temp/git/matrix-bifrost
> tsc

src/xmppjs/XJSConnection.ts:1:25 - error TS2307: Cannot find module '@xmpp/component-core' or its corresponding type declarations.

1 import {Component} from "@xmpp/component-core";
                          ~~~~~~~~~~~~~~~~~~~~~~

src/xmppjs/XJSConnection.ts:2:23 - error TS2307: Cannot find module '@xmpp/reconnect' or its corresponding type declarations.

2 import Reconnect from "@xmpp/reconnect";
                        ~~~~~~~~~~~~~~~~~


Found 2 errors.

 ERROR  Command failed with exit code 2.
```

**To Reproduce**
```
npm i
npm run build
```

**Expected behavior**
Packages with all required type declarations are provided in `package.json`.

**Desktop (please complete the following information):**
 - OS: Debian 11
 - npm: 6.14.8

--- a/changelog.d/192.bugfix
+++ b/changelog.d/192.bugfix
@@ -1,0 +1,1 @@
+- Add @xmpp/component-core and @xmpp/reconnect dependencies so the project is installed well with pnpm (#173). Thanks to @bodqhrohro

--- a/package.json
+++ b/package.json
@@ -26,7 +26,9 @@
   "homepage": "https://github.com/matrix-org/matrix-bifrost",
   "dependencies": {
     "@xmpp/component": "^0.11.0",
+    "@xmpp/component-core": "^0.11.0",
     "@xmpp/jid": "^0.11.0",
+    "@xmpp/reconnect": "^0.11.0",
     "@xmpp/xml": "^0.11.0",
     "fast-xml-parser": "^3.17.4",
     "he": "^1.2.0",
@@ -46,7 +48,6 @@
     "xmpp.js": "^0.6.1"
   },
   "devDependencies": {
-    "@types/pg": "^7.14.5",
     "@istanbuljs/nyc-config-typescript": "^1.0.1",
     "@types/chai": "^4.2.12",
     "@types/he": "^1.1.1",
@@ -54,6 +55,7 @@
     "@types/mocha": "^8.0.3",
     "@types/nedb": "^1.8.11",
     "@types/node": "^12",
+    "@types/pg": "^7.14.5",
     "@types/request-promise-native": "^1.0.17",
     "@types/xmpp__jid": "^1.3.1",
     "@types/xmpp__xml": "^0.6.1",
@@ -63,8 +65,8 @@
     "nedb": "^1.8.0",
     "nyc": "^15.1.0",
     "ts-node": "^9.0",
-    "typescript": "^4.0.3",
-    "tslint": "^6.1.3"
+    "tslint": "^6.1.3",
+    "typescript": "^4.0.3"
   },
   "optionalPeerDependencies": {
     "node-purple": "matrix-org/node-purple"

